src/algebra/algebra/basic.lean

```diff
@@ -758,7 +758,7 @@ end division_ring
 end alg_hom
 
 @[simp] lemma rat.smul_one_eq_coe {A : Type*} [division_ring A] [algebra ℚ A] (m : ℚ) :
-  m • (1 : A) = ↑m :=
+  @@has_smul.smul algebra.to_has_smul m (1 : A) = ↑m :=
```

Comment:
This is kind of a weird lemma at the moment because the `[algebra ℚ A]` parameter means it's safe against diamonds in `algebra ℚ A` instances. But the whole point of the PR is to fix those diamonds. Moreover the new `rat.smul_division_ring` instance also applies here, meaning we actually have two non-defeq `has_smul ℚ A` instances in the lemma statement. In the end, I decided on this form for the lemma to match closest to what it used to be, even though the diamonds that required the `[algebra ℚ A]` parameter are solved.

Reply:
We can always clean this up later; I imagine we avoided some similar choices for the nat and int algebras